src/Tmds.Ssh/KerberosCredential.cs

```diff
@@ -0,0 +1,327 @@
+    /// forwardable ticket required for delegation. When using a cached ticket, make sure that when the ticket was
+    /// retrieved it was retrieved with the forwardable flag. If the ticket is not forwardable, the authentication will
+    /// still work but the ticket will not be delegated.
+    /// </remarks>
+    /// <param name="credential">The credentials to use for the Kerberos authentication exchange. Set to null to use a cached ticket.</param>
+    /// <param name="delegateCredential">Allows the SSH server to delegate the user on remote systems.</param>
+    /// <param name="serviceName">Override the service principal name (SPN), default uses the <c>host/<SshClientSettings.Host>.</c></param>
+    public KerberosCredential(NetworkCredential? credential = null, bool delegateCredential = false, string? serviceName = null)
+    {
+        if (!string.IsNullOrWhiteSpace(credential?.UserName) && string.IsNullOrWhiteSpace(credential?.Password))
+        {
+            throw new ArgumentException("credential Password cannot be null or an empty string.", nameof(credential));
+        }
+        _kerberosCredential = credential ?? CredentialCache.DefaultNetworkCredentials;
+        _delegateCredential = delegateCredential;
+        _serviceName = serviceName;
+    }
+
+    internal async Task<bool> TryAuthenticate(SshConnection connection, ILogger logger, SshClientSettings settings, SshConnectionInfo connectionInfo, CancellationToken ct)
+    {
+        // RFC uses hostbased SPN format "service@host" but Windows SSPI needs the service/host format.
+        // .NET converts this format to the hostbased format expected by GSSAPI for us.
+        string spn = string.IsNullOrEmpty(_serviceName) ? $"host/{connectionInfo.Host}" : _serviceName;
+
+        // The SSH messages must have a username value which maps to the target user we want to login as. We use the
+        // client supplied username as the target user. The Kerberos principal credential is only used for the
+        // authentication stage that happens next.
+        string userName = settings.UserName;
+        logger.AuthenticationMethodGssapiWithMic(userName, _kerberosCredential.UserName, spn, _delegateCredential);
+
+        bool isOidSuccess = await TryStageOid(connection, logger, userName, ct).ConfigureAwait(false);
+        if (!isOidSuccess)
+        {
+            return false;
+        }
+
+        var negotiateOptions = new NegotiateAuthenticationClientOptions()
+        {
+            AllowedImpersonationLevel = _delegateCredential ? TokenImpersonationLevel.Delegation : TokenImpersonationLevel.Impersonation,
+            Credential = _kerberosCredential,
+            Package = "Kerberos",
+            // While only Sign is needed we need to set EncryptAndSign for
+            // Windows client support. Sign only will pass in SECQOP_WRAP_NO_ENCRYPT
+            // to MakeSignature which fails.
+            // https://github.com/dotnet/runtime/issues/103461
```

Comment:
hi @jborean93, the issue you referenced here is closed. Is there some change we can/should make for .NET9+?

Reply:
With the fix we can change it to `RequiredProtectionLevel = ProtectionLevel.Sign;`. but I don't think it ended up being backported to net9 so the fix is only present in net10. The PR which fixed it https://github.com/dotnet/runtime/pull/105605 was merged after net9 was created and I don't see any reference to a backport.

Reply:
Thanks for the input! I assumed it was fixed in .NET 9 based on the issue milestone.
We can add the change under an `#if NET10_0_OR_GREATER`.